lib/smack/SmackRep.cpp

```diff
@@ -834,8 +836,14 @@ const Expr* SmackRep::cmp(const llvm::ConstantExpr* CE) {
 }
 
 const Expr* SmackRep::cmp(unsigned predicate, const llvm::Value* lhs, const llvm::Value* rhs, bool isUnsigned) {
-  std::string fn = Naming::CMPINST_TABLE.at(predicate);
-  return Expr::fn(opName(fn, {lhs->getType()}), expr(lhs, isUnsigned), expr(rhs, isUnsigned));
+  std::string fn = opName(Naming::CMPINST_TABLE.at(predicate), {lhs->getType()});
+  const Expr* e1 = expr(lhs, isUnsigned);
+  const Expr* e2 = expr(rhs, isUnsigned);
+  if (lhs->getType()->isFloatingPointTy())
```

Comment:
I don't understand why different expressions are generated for integers vs floating-points. Please clarify. Thx!

Reply:
I still don't understand why different expressions are generated for ints and different for floating-point. Please comment.

Reply:
I see, thanks for the clarification. So why are then floating-point and integer comparisons not implemented uniformly? Maybe this is just a temporary fix, which is fine, but I would guess that eventually we should have two versions of floating-point comparisons as well, which would avoid the introduction of this if-then-else instruction. Is that the plan?